### ISSUE TITLE
[Bug] LoRA weights causing repetitive output (Token 27706) and EOS mismatch on Llama-7B

### ISSUE THREAD

**Issue by Emile-Jn** [ba0z1/Linguistic-Steganalysis-with-LLMs#2]
Hello, when I run the code in your test.ipynb, the LLM produces repetitive, nonsensical token generation, such as:

```
steganographic]{']{']{']{'
yes]{']{']{']{']{']{']{'
steganographic]{']{']{']{'
steganographic]{']{']{']{'
steganographic]{']{']{']{'
```

However, when I switch use_lora to `False` to use the base model without the fine-tuned weights, I don't see this behaviour. I see outputs like:

```
Yes, it is.  ### Question
No.  ### Question: Is
Yes.  ### Question: Is
Yes.  ### Question: Is
Yes  ### Question: Is the
```

When LoRA weights are used, the model consistently generates the sequence ]{']{']{']{ (Token ID: 27706) regardless of the input.

There appears to be a conflict between the tokenizer and the model's expected special tokens. The tokenizer's EOS is 0, while the model's EOS is 1. Note that the model never seems to emit this EOS token, and always reaches the max token limit, which is set to 10. Did the tokenizer or model get its vocab modified during fine-tuning?

It's not clear to me if the strange output is due to the tokenizer mismatch, or if a potential weight over-saturation in the adapter causes the model to enter a probability loop on Token 27706.

It's not clear which prompt template should be used for each checkpoint, out of the 8 prompt templates that you mention in your paper.

**Questions:** Did you encounter this type of output in your experiments? Which prompt template was used for each checkpoint?

Also, you provide code for training, but not the code that you used to run inference on the test set to obtain the results that you mention in your paper. It would be very useful to see all of your code.

Thank you very much,
Emile
